packages/ai-bot/lib/queries.ts

```diff
@@ -8,18 +8,20 @@ import {
   Expression,
 } from '@cardstack/runtime-common';
 
-export async function releaseLock(pgAdapter: PgAdapter, eventId: string) {
-  await query(pgAdapter, [
-    `DELETE FROM ai_bot_event_processing WHERE event_id_being_processed = `,
-    param(eventId),
-  ]);
-}
-
 export async function acquireLock(
   pgAdapter: PgAdapter,
   eventId: string,
   aiBotInstanceId: string,
 ): Promise<boolean> {
+  let existingLockRows = await query(pgAdapter, [
+    `SELECT 1 FROM ai_bot_event_processing WHERE event_id_being_processed = `,
+    param(eventId),
+  ]);
```

Comment:
This doesn’t need `WHERE completed_at IS NULL`?

Reply:
No, because we want to skip the event if it another process is working on it but has not completed it OR if any process worked on it and completed it.